### ISSUE TITLE
Calendar cannot be shared to a group 

### ISSUE THREAD

**Issue by dirkpape** [owncloud/core#23699]
### Steps to reproduce
1. login as user1 and share a calendar to a group of users containing user2, but not to the user2 himself
2. subscribe to the caldav service as the user2
3. see what calendars you can use

### Expected behaviour
calendar should be visible for user2

### Actual behaviour
calendar is not visible for user2.
But it is visible if shared directly (not via group) with user2.

### Server configuration
**Operating system**:
Ubuntu 14.4 LTE

**Web server:**
Server version: Apache/2.4.7 (Ubuntu)
Server built:   Jan 14 2016 17:45:23

**Database:**
mysql  Ver 14.14 Distrib 5.5.47, for debian-linux-gnu (x86_64) using readline 6.3

**PHP version:**
7.0.4

**ownCloud version:** (see ownCloud admin page)
9.0.1RC

**Calendar version:** (see ownCloud apps page)
1.0

**Updated from an older ownCloud or fresh install:**
updated from 8.2.3

**Signing status (ownCloud 9.0 and above):**

No errors have been found.

**List of activated apps:**

Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.1.0.0
  - dav: 0.1.6
  - direct_menu: true
  - documents: true
  - encryption: 1.2.0
  - external: 1.2
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_mv: true
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - galleryplus: true
  - logreader: true
  - news: 8.0.0
  - notifications: 0.2.3
  - ownnote: true
  - polls: 0.7.0
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0

Disabled:
  - gallery
  - user_external
  - user_ldap

**The content of config/config.php:**

{
    "system": {
        "instanceid": "518cbe88de454",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "xxxxx",
        "dbtype": "mysql",
        "version": "9.0.1.1",
        "dbname": "xxxxxxx",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "loglevel": 2,
        "forcessl": true,
        "forceSSLforSubdomains": false,
        "default_language": "de_DE",
        "enable_previews": true,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "trusted_domains": [
            "xxx"
        ],
        "mail_smtpmode": "php",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "noreply",
        "mail_domain": "xxxxx",
        "mail_smtphost": "localhost",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "custom_csp_policy": "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *",
        "maintenance": false,
        "enable_avatars": true,
        "appstore.experimental.enabled": true,
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0
        },
        "trashbin_retention_obligation": "auto",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

**Are you using external storage, if yes which one:** local/smb/sftp/...

no

**Are you using encryption:** yes/no

no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

no

### Client configuration
**Browser:**
Safari or Firefox

**Operating system:**
Macos X 10.11.4

**CalDAV-clients:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by RealRancor** [comment]
There have been some fixes for group sharing in https://github.com/owncloud/core/pull/23342 which gets shipped with oC 9.0.1.

**Comment by DeepDiver1975** [comment]
yes - will be fixed with 9.0.1

**Comment by alejandrofcarrera** [comment]
This issue appeared again few days ago, i have time today to attach all the info.

#### Steps to reproduce
* login as fst and share Meetings calendar to Developers group containing user2
* refresh website after did the share, the change is not visible
* login as user2 to review if the share was done
#### Expected behaviour
* Meetings calendar should be visible for user2
#### Actual behaviour
* calendar is not visible for user2, but it is visible if shared directly.
#### Server configuration
* CentOS 7 running latest Docker image of owncloud/server (Tag ID: 2e73a4f3de8c)
* MariaDB latest Docker image (Tag ID: cb0d69450b8e)
* ownCloud version: (see ownCloud admin page) - 10.0.8.5 (stable)
* Calendar version: (see ownCloud apps page) - 1.5.7
* The content of config/config.php:
```
<?php
$CONFIG = array (
  'datadirectory' => '/mnt/data/files',
  'updatechecker' => 'false',
  'upgrade.disable-web' => 'true',
  'skipUpdateCheck' => true,
  'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 =>
    array (
      'path' => '/var/www/owncloud/custom',
      'url' => '/custom',
      'writable' => true,
    ),
  ),
  'dbtype' => 'mysql',
  'dbhost' => 'ownsql:3306',
  'dbname' => 'owncloud',
  'dbuser' => 'root',
  'dbpassword' => '****************',
  'dbtableprefix' => 'oc_',
  'trusted_domains' =>
  array (
    0 => 'localhost',
  ),
  'passwordsalt' => '****************',
  'secret' => '****************',
  'overwrite.cli.url' => '****************',
  'version' => '10.0.8.5',
  'mysql.utf8mb4' => true,
  'logtimezone' => 'UTC',
  'installed' => true,
  'instanceid' => '****************',
  'redis' => array('host' => 'redis', 'port' => 6379),
  'filelocking.enabled' => 'true',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'loglevel' => '2',
  'default_language' => 'en',
  'htaccess.RewriteBase' => '/',
  'maintenance' => false,
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
);
```
#### Client configuration
* Browser: Chrome
* Operating system: Mac OS 10.13.4
#### Logs
``` **.**.**.*** - - [09/May/2018:10:04:10 +0000] "POST /remote.php/dav/calendars/fst/meetings/ HTTP/1.1" 200 647 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.139 Safari/537.36" ```
#### Owncloud logs
```{"reqId":"WLQsk2rM7p2nbjKzmE4h","level":3,"time":"2018-05-09T10:17:25+00:00","remoteAddr":"**.**.**.***","user":"fst","app":"OCP\\Share","method":"GET","url":"\/ocs\/v1.php\/apps\/files_sharing\/api\/v1\/sharees?format=json&search=deve&perPage=200&itemType=principals","message":"Sharing backend for principals not found"}```
#### Browser logs (payload sent)
```https://**********/remote.php/dav/calendars/fst/meetings/```
```<o:share xmlns:o="http://owncloud.org/ns"><o:set><d:href xmlns:d="DAV:">principal:principals/groups/Developers</d:href><o:summary>Meetings shared by fst</o:summary></o:set></o:share>```
